modules/gdscript/tests/test_completion.h

```diff
@@ -0,0 +1,201 @@
+
+	dir->list_dir_begin();
+	String next = dir->get_next();
+
+	while (!next.is_empty()) {
+		if (dir->current_is_dir()) {
+			if (next == "." || next == "..") {
+				next = dir->get_next();
+				continue;
+			}
+			test_directory(path.path_join(next));
+		} else {
+			if (next.ends_with(".gd")) {
+				Error err;
+				Ref<FileAccess> acc = FileAccess::open(path.path_join(next), FileAccess::READ, &err);
+
+				if (err != OK) {
+					next = dir->get_next();
+					continue;
+				}
+
+				String code = acc->get_as_utf8_string();
+				// Require pointer sentinel char in scripts.
+				CHECK(code.find_char(0xFFFF) != -1);
+
+				ConfigFile conf;
+				if (conf.load(path.path_join(next.get_basename() + ".cfg")) != OK) {
+					FAIL("No config file found.");
+				}
+
+#ifndef MODULE_MONO_ENABLED
+				if (conf.get_value("input", "cs", false)) {
+					next = dir->get_next();
+					continue;
+				}
+#endif
+
+				EditorSettings::get_singleton()->set_setting("text_editor/completion/use_single_quotes", conf.get_value("input", "use_single_quotes", false));
+
+				List<Dictionary> include;
+				to_dict_list(conf.get_value("result", "include", Array()), include);
+
+				List<Dictionary> exclude;
+				to_dict_list(conf.get_value("result", "exclude", Array()), exclude);
+
+				List<ScriptLanguage::CodeCompletionOption> options;
+				String call_hint;
+				bool forced;
+
+				Node *owner = nullptr;
+				if (dir->file_exists(next.get_basename() + ".tscn")) {
+					String project_path = "res://completion";
+					Ref<PackedScene> scene = ResourceLoader::load(project_path.path_join(next.get_basename() + ".tscn"), "PackedScene");
+					if (scene.is_valid()) {
+						owner = scene->instantiate();
+					}
+				}
+
+				GDScriptLanguage::get_singleton()->complete_code(code, path.path_join(next), owner, &options, forced, call_hint);
+				String contains_excluded = "";
+				for (ScriptLanguage::CodeCompletionOption &option : options) {
+					for (Dictionary &E : exclude) {
+						if (match_option(E, option)) {
+							contains_excluded = option.display;
+							break;
+						}
+					}
+					if (!contains_excluded.is_empty()) {
+						break;
+					}
+
+					for (Dictionary &E : include) {
+						if (match_option(E, option)) {
+							include.erase(E);
+							break;
+						}
+					}
+				}
+				CHECK_MESSAGE(contains_excluded.is_empty(), "Autocompletion suggests illegal option '", contains_excluded, "' for '", path.path_join(next), "'.");
+				CHECK(include.size() == 0);
+
+				String expected_call_hint = conf.get_value("result", "call_hint", call_hint);
+				bool expected_forced = conf.get_value("result", "forced", forced);
+
+				CHECK(expected_call_hint == call_hint);
+				CHECK(expected_forced == forced);
+
+				if (owner) {
+					owner->queue_free();
+				}
+			}
+		}
+		next = dir->get_next();
+	}
+}
+
+TEST_SUITE("[Modules][GDScript][Completion]") {
+	TEST_CASE("[Editor] Check suggestion list") {
+		// Set all editor settings that code completion relies on
```

Comment:
```suggestion
		// Set all editor settings that code completion relies on.
```
Per the [comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide).